Kernel/FileSystem/VirtualFileSystem.cpp

```diff
@@ -363,14 +363,15 @@ ErrorOr<void> VirtualFileSystem::mkdir(StringView path, mode_t mode, Custody& ba
     }
 
     RefPtr<Custody> parent_custody;
-    auto result = resolve_path(path, base, &parent_custody);
+    auto result = resolve_path_without_veil(path, base, &parent_custody);
     if (!result.is_error())
         return EEXIST;
     else if (!parent_custody)
         return result.release_error();
```

Comment:
I'm not sure if this is something we should care about, but in the case that this fails due to some permission issue or similar, we wouldn't return the normal ENOENT that unveil returns (essentially leaking that the file does exist, even though it wasn't unveiled)

Reply:
I am not sure how/if we can easily fix this. If `resolve_path_without_veil` fails e.g. with `ENOTDIR`, we would want to "hide" this if the respective path isn't unveiled. On the other hand, if it fails with `ENOMEM`, for example, we would want to obviously propagate that to the caller.

Do you have a good idea how to fix this?

Reply:
By the way, I think we have the same problem in `resolve_path` (which is used by almost every other function), which can also leak a possible `EACCESS` or whatever. I think I will just leave a `FIXME` for now.

Reply:
FIXME sounds good, as for "how can we solve this" we could probably just add a list of "masked errors" or "unmasked errors" and decide based on that if to return the actual error or the mask it with an ENOENT.